### PR TITLE
LPD-41466 Create playwright tests for "View" and "Permissions" permissions

### PR DESCRIPTION
**Pull request by thektan**
https://liferay.atlassian.net/browse/LPD-41466

Mainly for @kresimir-coko and @chestofwonders: An update on the branch I shared in the group chat. The most notable part is the `setupUserRoleAndLoginAsUser` function that now allows different actions to be selected. An example use would be:

```
await setupUserRoleAndLoginAsUser({
	apiHelpers,
	dataSetResourcePermissions: [
		{
			actions: ['Permissions', 'View'],
			name: 'Data Set',
		},
	],
	page,
	roleDefinePermissionsPage,
	rolePage,
	rolesPage,
});
```

All that's missing is the cleanup steps to delete the created role and user which we could do something similar to the `test.afterEach` in collecting the created role/user ids or ercs.

### CODE REVIEW

File: modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-admin/dataSets.spec.ts [hunk 1]
```diff
@@ -150,6 +163,131 @@ async function assertTableRowsCount(page, rowsCount) {
...
+			return Liferay.ThemeDisplay.getCompanyId();
+		});
+
+		dataSetUserRole = await apiHelpers.headlessAdminUser.postRole({
+			name: roleName,
+			rolePermissions: [
+				{
+					actionIds: ['VIEW_CONTROL_PANEL'],
+					primaryKey: companyId,
+					resourceName: '90',
```
Comment (dsanz) [inline modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-admin/dataSets.spec.ts:200]: why 90?

Comment (thektan) [reply modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-admin/dataSets.spec.ts:200]: Apparently it's the key for Portal, not sure how that number was chosen though:

https://github.com/liferay/liferay-portal/blob/master/portal-kernel/src/com/liferay/portal/kernel/util/PortletKeys.java#L132

**Comment by thektan** [comment]
Added @kresimir-coko's commit on top from https://github.com/liferay-frontend/liferay-portal/pull/4568/commits

I had a lot of issues with the tests not properly logging in as the admin user so I've added additional step to login for the moment: https://github.com/liferay-frontend/liferay-portal/pull/4558/commits/e721487949ec25caee1de85368f05ab4a46c5463#diff-2091b1c9d07194c1e29c22731a1a9944965c70916e131384a9f9dd09bf209830R302-R319

I'm still unsure why it happens (maybe only locally) but when I locally test, if a subsequent test runs after a test that logs in as a new user and back to admin, the new test doesn't log back in as admin. This change fixes it but definitely doesn't look great to keep. I'll need to look more into it.

**Comment by thektan** [comment]
ci:test:playwright

**Comment by thektan** [comment]
For the login issue, still unsure why this happens, but when the next test runs and the browser opens, the 2nd browser doesn't refresh as logged in at this step: https://github.com/liferay-frontend/liferay-portal/blob/master/modules/test/playwright/fixtures/loginTest.ts#L62

So so far my best option is having this `beforeEach` https://github.com/liferay-frontend/liferay-portal/pull/4558/commits/e721487949ec25caee1de85368f05ab4a46c5463#diff-2091b1c9d07194c1e29c22731a1a9944965c70916e131384a9f9dd09bf209830R302-R319 🫤

**Comment by dsanz** [comment]
@juanjofgliferay is adding these commits into the big PR we're preparing, so I'm closing this one

**Pull request closed without merging**